dev-support/ci/selective_ci_checks.bats

```diff
@@ -99,6 +99,39 @@ load bats-assert/load.bash
   assert_output -p needs-kubernetes-tests=false
 }
 
+@test "integration and unit: script change" {
+  run dev-support/ci/selective_ci_checks.sh c6850484f
```

Comment:
Could you help me understand why this specific SHA value chosen here?

It seems this specific SHA [c6850484f](https://github.com/apache/ozone/commits/c6850484f) dates back to Sep 2019?

Reply:
Sure.  I looked for a commit that only changed `_mvn_unit_report.sh`.  This file is shared between _unit_ and _integration_ checks, so it should trigger both.  It also triggers _bats_ due to change in a shell script, although there is no bats test for this specific script.  And _rat_ is needed for all changes.

Reply:
Interesting. Thanks for the explanation.

Sounds a bit like a "hack" to me, but acceptable.

A cleaner approach would be to, in some way, simply pass the relative file path to the project root to the `get_changed_files()` function in `selective_ci_checks.sh`?

Reply:
Sorry, I'm not sure I understand the proposed cleaner approach.  Can you please clarify?

Reply:
Yup. From my limited understanding of the script, it looks like the hash argument for `selective_ci_checks.sh` is only used here:

https://github.com/apache/ozone/blob/5843ab95/dev-support/ci/selective_ci_checks.sh#L512-L526

after that it is passed into here in `get_changed_files()`:

https://github.com/apache/ozone/blob/5843ab95/dev-support/ci/selective_ci_checks.sh#L62-L70

where the script generate the diff list of changed files from the given commit by running `git diff-tree --no-commit-id --name-only -r SHA~1 SHA` command.
So in the case of c6850484f, `CHANGED_FILES=hadoop-ozone/dev-support/checks/_mvn_unit_report.sh` (or in the case of multiple file change commits like 86a771dfe, the command gives relative file paths to the root separated by **line feeds**).

So wouldn't that be more straight forward to feed the file paths (a.k.a. `CHANGED_FILES`) directly to the `get_changed_files()` function rather than feeding a SHA? I'm guessing there was a historical reason for it? Some other functions are/were using the SHA for different purposes? But even then it is questionable if it is worth it to address this. This is more of a simple thought/question from me rather than a suggestion here.

Reply:
> [the SHA] is passed into here in `get_changed_files()`
> ...
> where the script generate the diff list of changed files from the given commit

vs.

> So wouldn't that be more straight forward to feed the file paths (a.k.a. `CHANGED_FILES`) directly to the `get_changed_files()` function rather than feeding a SHA?

The main task of `get_changed_files()` is generating the list of changed files.  (It also outputs it and handles the degenerate case of no changed files.)

So if we passed the list of changed files to this function, what would be its remaining job?  We'd also need some other function or code block to calculate the list of changed files, as the script's input is a SHA (normally the PR merge commit).

I think it's better to exercise it end-to-end by passing SHAs in the test, too.